lib/dor/text_extraction/cocina_updater.rb

```diff
@@ -62,14 +62,29 @@ def rename_document_pdf
         add_file_to_new_resource(new_filename)
       end
 
-      def add_file_to_resource(path)
-        if file_in_cocina?(path)
-          guard_overwrite(path)
+      def add_file(path)
+        guard_overwrite(path) if file_in_cocina?(path)
+
+        return unless path.exist? # we might have deleted the file if we are guarding against replacing it
+
+        resource = find_resource(path)
+
+        if resource
+          delete_file_from_resource(path, resource) if file_in_cocina?(path) # if it's in cocina, remove it first, so that we can replace it!
+          add_file_to_resource(path, resource)
         else
```

Comment:
this is the key change, basically we have to remove the file from cocina and then re-add it if a replacement.... previously for replacement, it wasn't touching cocina at all, thus guaranteeing the new files never replaced the old ones